### DISCUSSION TITLE
Query stuck in queue

### DISCUSSION THREAD

**Discussion by KaranpreetAnalytics** [getredash/redash#7226] [Q&A]
Hi Everyone,

Thanks for helping in advance!

I have a very strange issue in Redash v10 and v10.1 as well. My instance was created via AWS AMI on v8 and then upgraded to v10.0 and v10.1. My EC2 instance is running on c4.2xlarge config (8 cores and 16GB RAM).

I have a very specific query that goes in the queue and whatever I do be it restart upgrade, downgrade, force rebuild, or restart that query is always in the queue. The strangest part is all my workers are sitting idle and there is no query in the RQ status tab. I even checked the postgre internal database of Redash ( I have connected it to Redash itself for easy querying) and did not find that query in the queries tab. I found an event of a query action but could not figure out if that event got triggered because of the same query or not.

Datasource: Redshift

I have attached my query and some screenshots for the same. On the contrary, if I run a different query on the same data source and tables, I get the result and the issue is specific to this query specifically.

`SELECT DISTINCT 
  globaluserid, 
  globalcustomerid, 
  CONCAT(CAST(EXTRACT(week FROM time) AS VARCHAR), '-', CAST(EXTRACT(year FROM time) AS VARCHAR)) AS weekyear
FROM aep_enterprise_prod.cja_all_events a
LEFT JOIN aep_enterprise_prod.cja_users b ON a.user_id = b.user_id
WHERE event_table_name LIKE 'active_users_company'
  AND globaluserid IS NOT NULL
  AND globalcustomerid IS NOT NULL
LIMIT 100;`

![image](https://github.com/user-attachments/assets/e00a79b5-8625-4feb-a63d-7b4074bb6b53)

![image](https://github.com/user-attachments/assets/df19e916-179f-4325-81d6-e631ae5c9bcc)

PS: I have tried to run this query via multiple users and systems and the issue persists.

**Comment by gaecoli** [comment]
You should check redash's worker service is running?

**Reply by KaranpreetAnalytics** [reply]
It started running after I upgraded config to c4.4xlarge. Thanks for the help!
